Setting the `format` flag within a custom `lychee.toml` config file does not work

Hello,
I'm just trying to see if I can define the format inside my custom `lychee.toml` config file, but I found a couple of issues:

With the following config

```toml
format = "json"
```

I get the following error message:
```bash
   [ERROR] Error while loading config: Cannot load configuration file `./lychee.toml`: Failed to parse configuration file

Caused by:
    TOML parse error at line 38, column 10
       |
    38 | format = "json"
       |          ^^^^^^
    unknown variant `json`, expected one of `Compact`, `Detailed`, `Json`, `Markdown`, `Raw`
```


So then, I decided to change from `json` to `Json`, but it does not work and the output is shown in the default format `compact`.

--- a/fixtures/configs/format.toml
+++ b/fixtures/configs/format.toml
@@ -1,0 +1,1 @@
+format = "json"

--- a/lychee-bin/src/options.rs
+++ b/lychee-bin/src/options.rs
@@ -46,9 +46,10 @@ const TIMEOUT_STR: &str = concatcp!(DEFAULT_TIMEOUT_SECS);
 const RETRY_WAIT_TIME_STR: &str = concatcp!(DEFAULT_RETRY_WAIT_TIME_SECS);
 
 /// The format to use for the final status report
-#[derive(Debug, Deserialize, Default, Clone, Display, EnumIter, VariantNames)]
+#[derive(Debug, Deserialize, Default, Clone, Display, EnumIter, VariantNames, PartialEq)]
 #[non_exhaustive]
 #[strum(serialize_all = "snake_case")]
+#[serde(rename_all = "snake_case")]
 pub(crate) enum StatsFormat {
     #[default]
     Compact,
@@ -544,6 +545,7 @@ impl Config {
             exclude_link_local: false;
             exclude_loopback: false;
             exclude_mail: false;
+            format: StatsFormat::default();
             remap: Vec::<String>::new();
             fallback_extensions: Vec::<String>::new();
             header: Vec::<String>::new();

--- a/lychee-bin/tests/cli.rs
+++ b/lychee-bin/tests/cli.rs
@@ -1723,4 +1723,26 @@ mod cli {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_json_format_in_config() -> Result<()> {
+        let mock_server = mock_server!(StatusCode::OK);
+        let config = fixtures_path().join("configs").join("format.toml");
+        let mut cmd = main_command();
+        cmd.arg("--config")
+            .arg(config)
+            .arg("-")
+            .write_stdin(mock_server.uri())
+            .env_clear()
+            .assert()
+            .success();
+
+        // Check that the output is in JSON format
+        let output = cmd.output().unwrap();
+        let output = std::str::from_utf8(&output.stdout).unwrap();
+        let json: serde_json::Value = serde_json::from_str(output)?;
+        assert_eq!(json["total"], 1);
+
+        Ok(())
+    }
 }

